core/tests/test_handlers.py

```diff
@@ -714,3 +566,176 @@ def test_async_exception_cancelled_sync(event_loop):
 
     # The callback wasn't called
     on_result.assert_not_called()
+
+
+######################################################################
+# 2025-02: Handlers deprecated in 0.5.0
+######################################################################
```

Comment:
I think it's better not to move things to the bottom of the file when we deprecate them. Not only does this make the PR more difficult to read, it also breaks `git blame`.

Deprecated code can still be found by searching for "DeprecationWarning".